### DISCUSSION TITLE
how could I obtain all Microsoft365 and security and compliance reports (together)

### DISCUSSION THREAD

**Discussion by vanessav5** [silverhack/monkey365#110] [General]
Hi,

I have reviewed the documentation and I can see this report.  How could I obtain all Microsoft 365 reports together, including Security and Compliance.  I've generated every report separately, but I don't know how to create security and compliance, and later all reports together.

![monkey 365 complete report](https://github.com/user-attachments/assets/78455102-7e24-41b6-bde4-3d1d2b755349)

I'd like to create a complete report.  I appreciate your help.

Regards,

Vanessa V.

**Comment by silverhack** [comment]
Hi @vanessav5, 

Your tenant should have a valid license for Microsoft Purview. Without that license, Monkey365 will be unable to fetch data for obvious reasons. More information [here](https://learn.microsoft.com/en-us/purview/purview-compliance-portal-permissions).

and regarding getting all reports in one html file, you can populate the **--Analysis** flag with [multiple values](https://silverhack.github.io/monkey365/configuration/microsoft365-options/). To avoid misspelling commands, please make use of TAB-completion in PowerShell.

`
Invoke-Monkey365 -Instance Microsoft365 -Analysis ExchangeOnline,Microsoft365,MicrosoftTeams....
`

Cheers,

**Reply by vanessav5** [reply]
Hi @silverhack 

perfect, I have the complete report. 👍 

Thanks

Vanessa

**Reply by jrsantiago-osc** [reply]
@vanessav5 , what did you use for the whole report, both Azure and M365.  Besides the Purview licensing issues of course

**Comment by jrsantiago-osc** [comment]
Hi @silverhack , first off, love the tool.  As far as the reporting goes, how would you combine both the Azure and M365 into one report?  That sample report is about as perfect as it gets, and EXACTLY what I am looking for using the tool.  Thanks again

**Reply by vanessav5** [reply]
hi, well the complete line is:

$assets = Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect ExchangeOnline,MicrosoftTeams,AdminPortal,Purview,SharePointOnline

But you need Purview license to show this report..

Regards,

Vanessa

**Reply by jrsantiago-osc** [reply]
Thanks for the response.  When I try and use the -collect switch, it doesn't match a parameter unfortunately.  Even trying with tab completion, it just doesn't seem to be an actual switch.  And does that include both the Azure dropdown and Microsoft365 dropdown, like the sample report?  Looking to try and do everything all in one.

**Reply by vanessav5** [reply]
Hi @jrsantiago-osc 

No, one report is for Microsoft365 with Instance parameter and other report is for Azure.  @silverhack is it possible obtain the complete report for Azure and M365 together with all Microsoft 365 resources?

I appreciate your answer.

Regards,


Vanessa

**Comment by vanessav5** [comment]
$assets =  Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect ExchangeOnline,SharePointOnline,Purview,MicrosoftTeams

$assets =  Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Azure -Collect All

These are my commands.

regards,

Vanessa
